File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int i = 0; int j = nums.length-1;
        int idx = -1;
        while (i <= j) {
            int mid = (i+j)/2;
            
            if (nums[mid] == target) {
                return mid;
            }
            else if (target > nums[mid]) {
                i = mid +1;
            }
            else {
                j = mid -1;
            }
        }
        
        if (idx == -1) {
            if (j<=i) {
                return i;
            }
            else {
                return j;
            }
        }
        return 0;
    }
}

File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        
        int i = 0;
        int j = nums.size()-1;
        int mid;
        
        while(i <= j){
            mid = (i+j)/2;
            if(nums[mid] == target){
                return mid;
            }
            else if(target > nums[mid]){
                i = mid+1;
            }
            else {
                j = mid-1;
            }
        }
        if(target > nums[mid]){
            return mid+1; 
        }
        else{
            return mid;
        }
    }
};